contributingGuides/TS_STYLE.md

```diff
@@ -0,0 +1,335 @@
+   * @param age
+   * @returns Whether the person is a legal drinking age or nots
+   */
+  ```
+
+<a name="convension-proptypes-and-defaultprops"></a><a name="1.11"></a>
+
+- [1.11](#convension-proptypes-and-defaultprops) **`propTypes` and `defaultProps`**: Do not use them. Use object destructing to assign default values if necessary.
+
+  > Refer to [the propTypes Migration Table](./PROPTYPES_CONVERSION_TABLE.md) on how to type props based on existing `propTypes`.
+
+  ```tsx
+  type GreetingProps = {
+    greeting: string;
+    name: string;
+  };
+
+  function Greeting({ greeting = "hello", name = "world" }: ComponentProps) {
+    <Text>{`${greeting}, ${name}`}</Text>;
+  }
+  ```
+
+<a name="convension-utility-types"></a><a name="1.12"></a>
+
+- [1.12](#convension-utility-types) **Utility Types**: Use types from [TypeScript utility types](https://www.typescriptlang.org/docs/handbook/utility-types.html) and [`type-fest`](https://github.com/sindresorhus/type-fest) when possible.
+
+  ```ts
+  type Foo = {
+    bar: string;
+  };
+
+  // good
+  type ReadOnlyFoo = Readonly<Foo>;
+
+  // bad
+  type ReadOnlyFoo = {
+    readonly [Property in keyof Foo]: Foo[Property];
+  };
+  ```
+
+<a name="convension-object-type"></a><a name="1.13"></a>
+
+- [1.13](#convension-object-type) **`object`**: Don't use `object` type.
+
+> Why? `object` refers to "any non-primitive type," not "any object". Typing "any non-primitive value" is not commonly needed.
+
+```ts
+// bad
+const foo: object = [1, 2, 3]; // TypeScript does not error
+```
+
+If you know that the type of data is an object but don't know what properties or values it has beforehand, use `Record<string, unknown>`.
+
+<a name="convension-export-prop-types"></a><a name="1.14"></a>
+
+- [1.14](#convension-export-prop-types) **Prop Types**: Define and export prop types for components. Use exported prop types instead of grabbing the prop type from a component.
+
+> Why? Exporting prop types aids reusability.
+
+```tsx
+// MyComponent.tsx
+export type MyComponentProps = {
+  foo: string;
+};
+
+export default function MyComponent({ foo }: MyComponentProps) {
+  return <Text>{foo}</Text>;
+}
+
+// bad
+import { ComponentProps } from "React";
+import MyComponent from "./MyComponent";
+type MyComponentProps = ComponentProps<typeof MyComponent>;
+
+// good
+import MyComponent, { MyComponentProps } from "./MyComponent";
+```
+
+## Communication Items
+
+> Comment in the `#expensify-open-source` Slack channel if any of the following situations are encountered. Each comment should be prefixed with `TS ATTENTION:`. Internal engineers will access each situation and prescribe solutions to each case. Internal engineers should refer to general solutions to each situation that follows each list item.
+
+- I think types definitions in a third party library is incomplete or incorrect
+
+  When the library indeed contains incorrect type definitions and it cannot be updated, use module argumentation to correct them.
```

Comment:
```suggestion
  When the library indeed contains incorrect type definitions and it cannot be updated, use module augmentation to correct them.
```

Maybe let's also add a note to augmentate modules in `global.d.ts` file, wdyt @hayata-suenaga @fabioh8010?

Reply:
@blazejkustra that's a good idea 👍 changed accordingly